src/auth/bearer/fake/fake.cc

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2014, Treehouse Networks Ltd. New Zealand
```

Comment:
> Since the "copyright in question" is fully retained in CREDITS, we do not need it in this source file to satisfy the BSD license. The BSD license does not say "you must retain this notice in a file called foo/bar.cc".

We absolutely do. The CREDITS entry is for both our bookeeping and to satisfy the clause-2 requirement for attaching the licence to any built binaries. The clause-1 requires both copyright and licence to be left untouched, as-is in the code file to which they apply. Every. Single. File.

> Most of the leftover copyrights were grandfathered in because we did not have the energy to deal with them, not because something makes them a good idea that we should propagate.

Absolutely wrong. And I say that as I'm still doing the legwork to clear the legacy entries up. The legacy licence are being left as-is because that is the legal requirement imposed by those license terms. We have to do that, get written permission from the author(s), or completely replace the code it applies to. The standalone copyright statements (not part of an imported licence) are removed only with authors written permission to move it to CONTRIBUTORS. We were lucky that almost everyone was contactable, and agreed. But that CONTRIBUTORS move is only possible for standalone C statements not ones enmeshed with licence text.

> FWIW, I have removed or not added such statements from library-like code that Factory has contributed to Squid from other projects throughout the years. That does not prove me right, but at least I can say that I am eating my dog food (and intend to continue to do so).

I suggest you make sure that Duane or whoever is in charge of Factory are aware of that and keep some evidence to cover yourself, ASAP. Erasing either copyright statement or licence text from a file of code when one is not the author and copyright holder, nor has written permission to do so is outright illegal in any country subject to the Bourne Convention. Other laws may also be applicable depending on ones location. I'm only 100% sure the BC applies to me, but I think it applies in USA as well so be careful brother.

Since it is clear we are deadlocked on this for now I'm going to drop the helper and continue with the technical fixes in the morrow.